source/extensions/health_checkers/http/health_checker_impl.cc

```diff
@@ -81,6 +81,12 @@ HttpHealthCheckerImpl::HttpHealthCheckerImpl(const Cluster& cluster,
     service_name_matcher_.emplace(config.http_health_check().service_name_matcher());
   }
 
+  if (!Http::HeaderUtility::headerValueIsValid(path_) ||
+      !Http::HeaderUtility::headerValueIsValid(host_value_)) {
+    throw EnvoyException(fmt::format("the path {} or host name {} is not a valid header value.",
```

Comment:
nit: prefer separating to different if statements, and giving an explicit error to the field that causes the problem.
Also instead of throw, please use `THROW_IF_STATUS_NOT_OK/THROW_IF_NOT_OK`.